src/modules/adaptors/Compound/V3/SupplyAdaptor.sol

```diff
@@ -0,0 +1,177 @@
+        bytes memory adaptorData,
+        bytes memory configurationData
+    ) public view override returns (uint256) {
+        bool isLiquid = abi.decode(configurationData, (bool));
+        if (isLiquid) {
+            IComet comet = abi.decode(adaptorData, (IComet));
+            uint256 baseSupplied = comet.balanceOf(msg.sender);
+            uint256 baseLiquid = comet.baseToken().balanceOf(address(comet));
+            return baseSupplied > baseLiquid ? baseLiquid : baseSupplied;
+        } else return 0;
+    }
+
+    /**
+     * @notice Returns the balance of comet base token.
+     */
+    function balanceOf(bytes memory adaptorData) public view override returns (uint256) {
+        IComet comet = abi.decode(adaptorData, (IComet));
+        return comet.balanceOf(msg.sender);
+    }
+
+    /**
+     * @notice Returns `comet.baseToken()`
+     */
+    function assetOf(bytes memory adaptorData) public view override returns (ERC20) {
+        IComet comet = abi.decode(adaptorData, (IComet));
+        return comet.baseToken();
+    }
+
+    /**
+     * @notice This adaptor returns collateral, and not debt.
+     */
+    function isDebt() public pure override returns (bool) {
+        return false;
+    }
+
+    //============================================ Strategist Functions ===========================================
+
+    /**
+     * @notice Allows strategist to supply base token to Compound V3.
+     */
+    function supplyBase(IComet comet, uint256 assets) external {
+        _verifyComet(comet);
+
+        ERC20 base = comet.baseToken();
+        assets = _maxAvailable(base, assets);
+        base.safeApprove(address(comet), assets);
+
+        comet.supply(address(base), assets);
+
+        _revokeExternalApproval(base, address(comet));
+    }
+
+    /**
+     * @notice Allows strategist to withdraw base token from Compound V3.
+     */
+    function withdrawBase(IComet comet, uint256 assets) external {
+        _verifyComet(comet);
+
+        ERC20 base = comet.baseToken();
+        uint256 baseSupplied = comet.balanceOf(address(this));
+        uint256 baseLiquid = base.balanceOf(address(comet));
+
+        // Cap withdraw amount to be baseSupplied so that a strategist can not accidentally open a borrow using this function.
+        if (assets > baseSupplied) assets = baseSupplied;
+
+        // Cap withdraw amount to what is liquid in comet.
+        if (assets > baseLiquid) assets = baseLiquid;
+
+        comet.withdraw(address(base), assets);
+    }
+
+    /**
+     * @notice Reverts if a Cellar is not setup to interact with a given Comet.
+     * @dev This function is only used in a delegate call context, hence why address(this) is used
+     *      to get the calling Cellar.
+     */
+    function _verifyComet(IComet comet) internal view {
+        uint256 cellarCodeSize;
+        address cellarAddress = address(this);
+        assembly {
+            cellarCodeSize := extcodesize(cellarAddress)
```

Comment:
this check is here so that if the compound v3 supply position is set as the holding position of the cellar, we dont try to call back to the cellar while the cellar is being deployed(which would revert) so the only time the code size will be zero is during deployment.

Reply:
Hmm, perhaps there's an extensible design setup for Sommelier Cellars v3 or something with this. Or maybe not. Either way, good to think about moving forward. Thanks!